### PR TITLE
[DEV-6756] Check es doc duplication management command

### PR DESCRIPTION
**Pull request by kwhickey**
**Description:**
Adding a management command to count doc-duplication statistics in an Elasticsearch index.

NOTE: This is a _utility_ management command to adding in data quality checks or troubleshooting/debugging. It can be used as-is, but has some performance caveats. It works, but could be improved upon if further use and specialization is desired.

**Technical details:**
- Uses one of a scroll or partition strategy to progress through all docs
- The former has client-side burden, while the latter has server-side burden
- These are useful if truly needing to get a count of duplication, but doing so comes at a lot of processing time (several hours likely)
    - Could be used for instance to provide assurance that an error in ES indexing or deletion didn't yield duplicate documents in an index

**Requirements for PR merge:**

1. [NA] Unit & integration tests updated
2. [NA] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
    - [NA] Frontend <OPTIONAL>
    - [NA] Operations <OPTIONAL>
    - [NA] Domain Expert <OPTIONAL>
4. [NA] Matview impact assessment completed
5. [NA] Frontend impact assessment completed
6. [NA] Data validation completed
7. [NA] Appropriate Operations ticket(s) created
8. [x] Jira Ticket [DEV-6756](https://federal-spending-transparency.atlassian.net/browse/DEV-6756):
    - [x] Link to this Pull-Request
    - [x] Performance evaluation of affected (API | Script | Download)
    - [x] Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

File: usaspending_api/etl/management/commands/check_es_doc_duplication.py
Comment (tony-sappe) [inline usaspending_api/etl/management/commands/check_es_doc_duplication.py:12]: Since this is a management command, it should be "script" instead of the module name. Also the convention is to use `logger` instead of `_log`.


```suggestion
logger = logging.getLogger("script")
```

File: usaspending_api/etl/management/commands/check_es_doc_duplication.py
Comment (tony-sappe) [inline usaspending_api/etl/management/commands/check_es_doc_duplication.py:41]: Is it necessary to name everything with a trailing underscore? This is a management command, the likelihood of it being imported by other code is incredibly small and discouraged.

File: usaspending_api/etl/management/commands/check_es_doc_duplication.py
Comment (tony-sappe) [inline usaspending_api/etl/management/commands/check_es_doc_duplication.py:133]:
```suggestion
        # Instantiate a numpy array with the number of docs' _id fields we expect to touch
```

File: usaspending_api/etl/management/commands/check_es_doc_duplication.py
Comment (tony-sappe) [inline usaspending_api/etl/management/commands/check_es_doc_duplication.py:70]: Could use the env var + global variable `ES_HOSTNAME` like the other scripts

File: usaspending_api/etl/management/commands/check_es_doc_duplication.py
Comment (tony-sappe) [inline usaspending_api/etl/management/commands/check_es_doc_duplication.py:135]:
```suggestion
        _log.info(f"Collecting _ids from all {doc_count:,} in scrolling batch queries of size {self._partition_size:,}")
```

**Review by tony-sappe — COMMENTED** [review]
Leaving a few comments. Since this is a nice-to-have management command they don't need to be blockers to merging

File: usaspending_api/etl/management/commands/check_es_doc_duplication.py
Comment (kwhickey) [reply usaspending_api/etl/management/commands/check_es_doc_duplication.py:41]: It's a good practice we should follow everywhere in our python code. If fields are intended to only be used within a class itself (aka "private" in an object-oriented sense), they should be marked via a naming convention as such.

Also generates IDE warnings (as if the underscore wasn't enough) when you are using a "private" field.

File: usaspending_api/etl/management/commands/check_es_doc_duplication.py
Comment (kwhickey) [reply usaspending_api/etl/management/commands/check_es_doc_duplication.py:70]: Left out for flexibility of how this is used - a bit more flexibility by not having to have env vars setup and pulled in via `settings.py` to run this. Could be future improvement if used regularly

File: usaspending_api/etl/management/commands/check_es_doc_duplication.py
Comment (kwhickey) [reply usaspending_api/etl/management/commands/check_es_doc_duplication.py:12]: I still don't think this is the best way to do this. 
By hard-coding this to a single logger named "script", there is no way to adjust log level of different modules executed _during_ the script run via the logging config in `settings.py`. 

The way this should be done is to have e.g. this (and other) management command modules all getting associated with the `"script"` handler by way of providing their module name as a logger in the `loggers` section of the logging config dict. If that becomes too verbose, that dict can be modified to pattern match (e.g. all modules that have `*.management.*` in their module namespace) to associate them with the `script` handler.

File: usaspending_api/etl/management/commands/check_es_doc_duplication.py
Comment (kwhickey) [reply usaspending_api/etl/management/commands/check_es_doc_duplication.py]: Fixed

File: usaspending_api/etl/management/commands/check_es_doc_duplication.py
Comment (kwhickey) [reply usaspending_api/etl/management/commands/check_es_doc_duplication.py]: Fixed

File: usaspending_api/etl/management/commands/check_es_doc_duplication.py
Comment (sethstoudenmier) [inline usaspending_api/etl/management/commands/check_es_doc_duplication.py:122]: Not worth blocking on, but you can add this check in the argument parser using `choices`: https://docs.python.org/3/library/argparse.html#choices

File: usaspending_api/etl/management/commands/check_es_doc_duplication.py
Comment (sethstoudenmier) [inline usaspending_api/etl/management/commands/check_es_doc_duplication.py:173]: This printing out of rows is really nice for the `scroll` approach and would be nice to have for the `partition` approach.

**Review by sethstoudenmier — APPROVED** [review]
LGTM and is REALLY nice to have! Tested locally by forcing a duplicate document. Both approaches picked up on the duplicate and reporting accurately. The table printout in `scroll` would be nice to have for `partition`, but won't block on it. Approving and leaving for you to merge if not worth adding that table view.

**Pull request merged**
